### PR TITLE
Finishing touches for the network crawler

### PR DESCRIPTION
**Pull request by ljedrz**
This PR is a set of small tweaks to the crawler, its docs and the related test. In addition, it now logs the average handshake time.

### CODE REVIEW

File: .crawler/src/known_network.rs [hunk 1]
```diff
@@ -66,42 +68,36 @@ impl fmt::Debug for NetworkSummary {
...
-            listening_addr,
-            state: None,
-            timestamp: None,
-            received_peer_sets: 0,
-            connection_failures: 0,
-        }
-    }
-
+    // Resets the node's values which determine whether the crawler should stay connected to it.
+    // note: it should be called when a node is disconnected from after it's been crawled successfully
```
Comment (niklaslong) [inline .crawler/src/known_network.rs:93]:
```suggestion
    // note: it should be called when a peer is disconnected from after it's been crawled successfully
```

Comment (niklaslong) [reply .crawler/src/known_network.rs:93]: Actually we use `node` throughout here, ignore this 👍

File: .crawler/src/known_network.rs [hunk 2]
```diff
@@ -307,6 +315,8 @@ impl KnownNetwork {
         let mut states = HashMap::with_capacity(nodes.len());
         let mut types = HashMap::with_capacity(nodes.len());
         let mut heights = HashMap::with_capacity(nodes.len());
+
+        let mut handshake_times = Vec::with_capacity(nodes.len());
```
Comment (niklaslong) [inline .crawler/src/known_network.rs:307]: Should we be using a bounded circular queue here instead?

Comment (ljedrz) [reply .crawler/src/known_network.rs:307]: no need, as this collection is created ad-hoc for the currently known nodes only

**Review by niklaslong — APPROVED** [review]

**Pull request merged**